component-processor/src/main/java/org/infinispan/component/processor/ComponentAnnotationProcessor.java

```diff
@@ -168,12 +169,13 @@ private void parseGeneratedSources(ModelBuilder modelBuilder) throws IOException
       }
 
       TypeElement moduleClassElement = elements().getTypeElement(moduleClassAndPackages.getKey());
-      if (moduleClassElement == null || moduleClassElement.getAnnotation(InfinispanModule.class) == null) {
+      InfinispanModule moduleAnnotation = moduleClassElement != null ?
+                                          getAnnotation(moduleClassElement, InfinispanModule.class) : null;
+      if (moduleAnnotation == null) {
          info(null, "Ignoring invalid module implementation %s", moduleImplementationName);
```

Comment:
I wonder if making this an error rather than an info would improve the experience. I do realize that once this happens here the build will certainly fail later for other module-related reasons, but maybe it would be good to ensure these errors are not silent.

Reply:
@anistor this is the old generated module file, and if the annotation processor can't read it, there is nothing the user can do to change that, so logging an error doesn't really improve the user experience.

A module implementation class can become invalid when switching branches or when working on the annotation processor itself, and in both scenarios logging an error and stopping the build so that the user adds "clean" to the command line would be just annoying.